docs_src/microservices/application/ContextAPI.md

```diff
@@ -1,108 +1,142 @@
+    NotificationClient() interfaces.NotificationClient
+    SubscriptionClient() interfaces.SubscriptionClient
+    DeviceServiceClient() interfaces.DeviceServiceClient
+    DeviceProfileClient() interfaces.DeviceProfileClient
+    DeviceClient() interfaces.DeviceClient
+    PushToCore(event dtos.Event) (common.BaseWithIdResponse, error)
+    GetDeviceResource(profileName string, resourceName string) (dtos.DeviceResource, error)
+    AddValue(key string, value string)
+    RemoveValue(key string)
+    GetValue(key string) (string, bool)
+    GetAllValues() map[string]string
+    ApplyValues(format string) (string, error)
 }
 ```
 
+## InputContentType()
+`InputContentType()` returns the content type of the data that initiated the pipeline execution. Only useful when the TargetType for the pipeline is []byte, otherwise the data with be the type specified by TargetType.
+
+## SetResponseData()
+`SetResponseData(data []byte)` sets the response data that will be returned to the trigger when pipeline execution is complete.
+
+## ResponseData()
+`ResponseData()` returns the data that will be returned to the trigger when pipeline execution is complete.
+
+## SetResponseContentType()
+`SetResponseContentType(string)` sets the content type that will be returned to the trigger when pipeline execution is complete.
+
+## ResponseContentType()
+`ResponseContentType()` returns the content type that will be returned to the trigger when pipeline execution is complete.
+
+## SetRetryData()
+`SetRetryData(data []byte)` set the data that is to be retried later as part of the Store and Forward capability.  Used when there was failure sending the data to an external source.
+
 ## Clients
 
-### LoggingClient
+### LoggingClient()
+
+Returns a `LoggingClient` to leverage logging libraries/service utilized throughout the EdgeX framework. The SDK has initialized everything so it can be used to log `Trace`, `Debug`, `Warn`, `Info`, and `Error` messages as appropriate. See [simple-filter-xml/main.go](https://github.com/edgexfoundry-holding/app-service-examples/blob/master/app-services/simple-filter-xml/main.go) for an example of how to use the `LoggingClient`.
 
-The `LoggingClient` exposed on the context is available to leverage logging libraries/service utilized throughout the EdgeX framework. The SDK has initialized everything so it can be used to log `Trace`, `Debug`, `Warn`, `Info`, and `Error` messages as appropriate. See [simple-filter-xml/main.go](https://github.com/edgexfoundry-holding/app-service-examples/blob/master/app-services/simple-filter-xml/main.go) for an example of how to use the `LoggingClient`.
+### EventClient()
 
-### EventClient 
+Returns an `EventClient` to leverage Core Data's `Event` API. See [interface definition](https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/clients/interfaces/event.go) for more details. This client is useful for querying events and is used by the [PushToCore](#pushtocore) convenience API described below.
```

Comment:
```suggestion
Returns an `EventClient` to leverage Core Data's `Event` API. See [interface definition](https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/clients/interfaces/event.go) for more details. This client is useful for querying events and is used by the [PushToCore](#pushtocore) convenience API described below. Note if Core Data is not specified in the Clients configuration, this will return nil.

```